File: src/com/jepl/lang/JEPLException.java
package com.jepl.lang;
import static com.jepl.lang.Interpreter.block;

public class JEPLException extends RuntimeException {
    public JEPLException(Throwable e){
        super(e);
        int b = block+1;

        System.err.println("Error [block " + b + "] - " + e);
        System.exit(-1);
    }
}

File: src/com/jepl/lang/JEPLException.kt
package com.jepl.lang

import com.google.gson.JsonObject

class JEPLException(e: Throwable, at: JsonObject? = null): Exception(e) {
    private var block = Interpreter.block+1
    private var throwable = e

    init {
        System.err.println("Error [block $block] - ${this.throwable.message}\nInterpreter says that something is wrong here:   ${at!!}\n\nJVM stacktrace:")
    }
}